sdk/iot/core/src/az_iot_telemetry.c

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "az_iot_hub_client.h"
+#include <az_contract_internal.h>
+#include <az_result.h>
+#include <az_span.h>
+
+static az_span telemetry_prop_delim = AZ_SPAN_LITERAL_FROM_INITIALIZED_BUFFER((uint8_t[]){ '?' });
+static az_span telemetry_prop_separator
+    = AZ_SPAN_LITERAL_FROM_INITIALIZED_BUFFER((uint8_t[]){ '&' });
+static az_span telemetry_null_terminator
+    = AZ_SPAN_LITERAL_FROM_INITIALIZED_BUFFER((uint8_t[]){ '\0' });
+static az_span telemetry_topic_prefix = AZ_SPAN_LITERAL_FROM_STR("devices/");
+static az_span telemetry_topic_modules_mid = AZ_SPAN_LITERAL_FROM_STR("/modules/");
+static az_span telemetry_topic_suffix = AZ_SPAN_LITERAL_FROM_STR("/messages/events/");
```

Comment:
Can we just have literal... Right now you are using flash for literal and 2 ints, for every const

Reply:
Also make sure every literal is with const.

---
In reply to: [389163783](https://github.com/CIPop/azure-sdk-for-c/pull/1#discussion_r389163783) [](ancestors = 389163783)

Reply:
but span has two additional fields right? Are they required to be in flash?

---
In reply to: [389163934](https://github.com/CIPop/azure-sdk-for-c/pull/1#discussion_r389163934) [](ancestors = 389163934,389163783)

Reply:
i see. in the case of the single character's I will use the `az_span_append_uint8`. Thanks for that. Also I'm going to move this to the PR to the actual sdk